sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -619,7 +642,11 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
 
     var table = inputTable
 
-    if (table.tableType != VIEW) {
+    if (table.tableType == VIEW) {
+      // Read view default database from table properties.
+      val viewDefaultDatabase = table.properties.get(VIEW_DEFAULT_DATABASE)
```

Comment:
We may not have VIEW_DEFAULT_DATABASE in the properties for views that are defined in older versions of spark, right?